discovery-infra/test_infra/helper_classes/nodes.py

```diff
@@ -149,3 +151,12 @@ def get_cluster_hostname(self, cluster_host_object):
         inventory = json.loads(cluster_host_object["inventory"])
         return inventory["hostname"]
 
+    def set_hostnames(self, cluster):
+        ipv6 = env_variables.get('ipv6')
+        if ipv6:
+            network_name = self.controller.params.libvirt_network_name
+            libvirt_nodes = utils.get_libvirt_nodes_from_tf_state(network_name, self.controller.tf.get_state())
+            utils.update_hosts(cluster.api_client, cluster.id, libvirt_nodes, update_hostnames=True)
```

Comment:
This workflow isn't clear to me at all. Why do the hostnames get updated in the case of ipv6?
Please add a description with comments

Reply:
OK.  I will add comment